app/models/PhoneNumberProvider.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace models;
+
+class PhoneNumberProvider
+{
+
+	protected $twilioClient;
+	protected $countryCode;
+	protected $phoneNumber;
+
+	public function __construct(\Services_Twilio $twilioClient, $countryCode)
+	{
+		$this->twilioClient	 = $twilioClient;
+		$this->countryCode	 = strtoupper($countryCode);
+	}
+
+	public function getPhoneNumber()
+	{
+		if (!$this->loadPhoneNumber()) {

```

Comment:
should perform exact type checking. For exampe:
`$number === null` or `!$this->hasPhoneNumber()` while `hasPhoneNumber()` return **boolean**